### PR TITLE
Bugfix FXIOS-9681 - Few constraints and scrollview are not working as expected for TrackingProtectionViewController

### PR DESCRIPTION
**Pull request by dicarobinho**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9681)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/21294)

## :bulb: Description
Fixed few constraints and scrollview for TrackingProtectionViewController modal

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift
Comment (cyndichin) [inline firefox-ios/Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift:70]: can we add this hardcoded value within a `struct UX`?

File: firefox-ios/Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift
Comment (cyndichin) [inline firefox-ios/Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift:73]: how does this look like in multitasking? do we want to use `shouldUseiPad` check instead?

File: firefox-ios/Client/Frontend/Browser/EnhancedTrackingProtection/SlideoverPresentationController.swift
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Browser/EnhancedTrackingProtection/SlideoverPresentationController.swift:16]: why are we adding additional implementation here for a legacy code?

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (cyndichin) [inline firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:280]: can we add tests for this class since its a new file?

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (cyndichin) [inline firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:16]: lets use our `FXFontStyles` to set this fonts

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (cyndichin) [inline firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:862]: should we add a TODO with a ticket here?

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (cyndichin) [inline firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:801]: same here in terms of adding TODO?

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (cyndichin) [inline firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:56]: the Redux architecture should replace these delegates, I believe. I also didn't see redux implemented for this VC. Assuming that it will be in a TODO, should we add that somewhere in this view controller?

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (cyndichin) [inline firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:145]: add this hardcoded values to a UX struct?

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (cyndichin) [inline firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:124]: could we separate this UI components in its own view and then instantiate them in this view controller. It can avoid this VC become too massive and decouple the views more.

File: firefox-ios/Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift
Comment (dicarobinho) [reply firefox-ios/Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift:73]: Not sure about this comment here but line 72 will be executed only for .phone cases and not for iPads.

File: firefox-ios/Client/Frontend/Browser/EnhancedTrackingProtection/SlideoverPresentationController.swift
Comment (dicarobinho) [reply firefox-ios/Client/Frontend/Browser/EnhancedTrackingProtection/SlideoverPresentationController.swift:16]: It has been renamed so we still want to keep the old delegate finish() method.

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (dicarobinho) [reply firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:862]: Nope, deleted.

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (dicarobinho) [reply firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:280]: For TrackingProtectionModel, or?

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (dicarobinho) [reply firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:862]: No, we don't.

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (dicarobinho) [reply firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:56]: Redux side is done by Stefan and will be merged soon with this controller.

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (dicarobinho) [reply firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:124]: Since the views are different one from each other, do you have a suggestion how this "decouple" should looks like? Thank you.

File: firefox-ios/Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift
Comment (dicarobinho) [reply firefox-ios/Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift:73]: As I tested, it is working well on multitasking / split view mode for iPad.

Comment (cyndichin) [reply firefox-ios/Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift:77]: okay, thanks for checking!

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (cyndichin) [reply firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:280]: for `TrackingProtectionViewController`

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (cyndichin) [reply firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:124]: I was thinking something along the line of how we have `MicrosurveyConfirmationView` and then its being used in `MicrosurveyViewController`. In this case, we can create a separate `TrackingProtectionConnectionDetailsView` and then call it in the view controller here.

Something like this, let me know if this makes sense. Happy to chat about it as well to clarify!
```
private lazy var connectionDetailsView: TrackingProtectionConnectionDetailsView = .build()
```

File: firefox-ios/Client/Frontend/Browser/EnhancedTrackingProtection/SlideoverPresentationController.swift
Comment (cyndichin) [reply firefox-ios/Client/Frontend/Browser/EnhancedTrackingProtection/SlideoverPresentationController.swift:16]: okay, thank you for the explanation!

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (petruSt24) [reply firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:280]: there are some tests added in the `TrackingProtectionStateTests` and more will come after this PR gets merged and we can connect the `TrackingProtectionViewController` with the redux actions and flows

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (dicarobinho) [reply firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:124]: but my suggestion: in order to not hold this PR only for a refactor, let me to create a ticket and put it as a TODO in the TrackingProtectionViewController file and it can be handled in future.

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift
Comment (cyndichin) [reply firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:281]: I was thinking at least memory leak test as a minimum, but this can be done in a separate PR so not blocking this one!

**Review by cyndichin — APPROVED** [review]
thanks @dicarobinho for making these updates! approving assuming that some of the other concerns will be done in future PRs.

**Review by dragosb01 — APPROVED** [review]

**Pull request merged**
